<?xml version="1.0"?>
<launch>
    <arg name="thread_num" default="8"/>
    <arg name="use_camera1" default="1"/>
    <!-- TRIANGULATION -->
    <node pkg="triangulation" type="triangulation.py" name="triangulation" output="screen">
        <rosparam command="load" file="$(find triangulation)/config/params_camera_matrix.yaml" />
        <param name="camera_frame" value="mono_link" /> <!-- camera frame topic -->
        <param name="camera_topic" value="/blackfly/image_color" /> <!-- camera feed topic -->
        <param name="odom" value="/navsat/odom" /> <!-- Odometry topic -->
        <param name="image_rate" value="1" /> <!-- frequency of collecting images (rate per second) -->
        <param name="triang_file" value="/home/autoronto/triangulation/test_node.csv" /> <!-- location for storing CSV file-->
        <param name="folder" value="/home/autoronto/triangulation/test_node/" /> <!--location for storing captured images -->
        <param name="collect_image" value="1" />
        <param name="output_append" value="" />
    </node>
    <!-- UNCOMPRESS IMAGES FROM ROSBAG -->
    <node pkg="image_transport" type="republish" name="cam1_republish" output="screen" if="$(arg use_camera1)"
      args="compressed in:=/blackfly/image_color raw out:=/blackfly/image_color"/>
</launch>
